recipes/cubicinterpolation/all/conanfile.py

```diff
@@ -0,0 +1,98 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class CubicInterpolationConan(ConanFile):
+    name = "cubicinterpolation"
+    homepage = "https://github.com/MaxSac/cubic_interpolation"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Leightweight interpolation library based on boost and eigen."
+    topics = ("interpolation", "splines", "cubic", "bicubic", "boost", "eigen3")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+    _source_subfolder = "source_subfolder"
+    exports_sources = ["CMakeLists.txt"]
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, "14")
+
+        minimum_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler), False
+        )
+        if not minimum_version:
+            self.output.warn(
+                "CubicInterpolation requires C++14. Your compiler is unknown. Assuming it supports C++14."
+            )
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                "CubicInterpolation requires C++14, which your compiler does not support."
+            )
+
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("This library is not compatible with Windows")
```

Comment:
Windows or Visual Studio only?

Reply:
I am not super literate in Windows as well as VS Code, but if I understand the logs correctly no other combination of Windows with another compiler, as well as VS Code with another OS is tested. 

If desired I could install VS Code on Linux and verify if the installation is possible there.

Reply:
VSCode is a text editor, not a compiler. Does it compile with MinGW gcc on Windows?

Reply:
To be honest, I didn't realize until just now that there was a difference between the two. The library doesn't use any OS specific functions, but I don't know what could cause Windows incompatibilities. 

In the [log](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/4695/22/cubicinterpolation/0.1.2//summary.json) it seems that no other combinations are tested. Is it possible that other compilers are tested when Windows is run with VS and in this case stopped too early? I don't have a Windows capable PC at the moment either. Is there a better way to find out, besides looking for a computer where I can test this combination?

Reply:
Actually, I'm pretty sure it compiles with MinGW.
I'm able to compile your library with Visual Studio and just few modifications:
- add an explicit `static_cast<long int>` of `y.rows()` and `y.cols()` in `return std::array<long int, 2>{y.rows(), y.cols()}` in `BicubicSplines.cxx`
- do use `not` keyword in `InterpolantBuilder.h`, seems that Visual Studio is not aware of this unusual keyword.
- math constants (e.g `M_LN2` in your library) are not defined by default with Visual Studio when you include `cmath`.
  You can add this in your CMakeLists.txt:
  ```cmake
  if(MSVC)
    target_compile_definitions(CubicInterpolation PRIVATE _USE_MATH_DEFINES)
  endif()
  ```
- Finally for shared lib, you'll need to export all symbols and install runtime in bin folder instead of lib:
  ```cmake
  if(MSVC)
    target_property(TARGET CubicInterpolation PROPERTY WINDOWS_EXPORT_ALL_SYMBOLS ON)
  endif()
  ```
  ```cmake
  install(TARGETS CubicInterpolation EXPORT CubicInterpolationTargets
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/CubicInterpolation
  )
  ```

Reply:
I'll submit a PR to your branch

Reply:
Thank you very much! 
Would it be ok for you if I create a library bug fix release that includes your changes, or is it good practice to create a conan patch?

Reply:
I've created a PR here for this recipe: https://github.com/conan-io/conan-center-index/pull/4695#issuecomment-789643134

It's fine to create a patch in the recipe for this version.